### PR TITLE
Automated cherry pick of #2140: Add expiry to kb session

### PR DESCRIPTION
**Pull request by rene-dekker**
Cherry pick of #2140 on release-v1.28.

#2140: Add expiry to kb session

### CODE REVIEW

**Review by Brian-McM — APPROVED** [review]

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and delete the branch after I've merged it.

**Comment by marvin-tigera** [comment]
I'm sorry but I failed delete the branch after merging the pull request.

**Pull request merged**
